File: vertx-module/src/test/java/org/jesperancinha/votd/FinanceVerticleTest.java
package org.jesperancinha.votd;

import io.vertx.core.Vertx;
import io.vertx.core.http.HttpMethod;
import io.vertx.ext.unit.Async;
import io.vertx.ext.unit.TestContext;
import io.vertx.ext.unit.junit.VertxUnitRunner;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

/**
 * Created by joao on 21-2-16.
 */
@RunWith(VertxUnitRunner.class)
public class FinanceVerticleTest {

    private Vertx vertx;

    @Before
    public void setUp(TestContext context){
        vertx = Vertx.vertx();
        vertx.deployVerticle(FinanceVerticle.class.getName(), context.asyncAssertSuccess());
    }

    @After
    public void tearDown(TestContext context){
        vertx.close();
    }

    @Test
    public void testMyApplication(TestContext context) {
        final Async async = context.async();

        vertx.createHttpClient().request(HttpMethod.GET,8080, "localhost", "/",
                response -> {
            response.map(body -> {
//                context.assertTrue(body.toString().contains("This is only the begining of cashing in"));
                async.complete();
                return null;
            });
        });
    }
}

File: vertx-module/src/test/kotlin/org/jesperancinha/votd/FinanceVerticleTest.kt
package org.jesperancinha.votd

import io.vertx.core.DeploymentOptions
import io.vertx.core.Vertx
import io.vertx.core.http.HttpMethod
import io.vertx.junit5.VertxExtension
import io.vertx.junit5.VertxTestContext
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

/**
 * Created by joao on 21-2-16.
 */
@ExtendWith(VertxExtension::class)
class FinanceVerticleTest constructor(
    context: VertxTestContext
) {

    init {
        context.completeNow()
    }

    private val vertx: Vertx by lazy {
        Vertx.vertx().also {
            it.deployVerticle(FinanceVerticle(), DeploymentOptions())
                .onComplete { result ->
                    if (result.succeeded()) {
                        println("Deployed: ${result.result()}")
                    } else {
                        println("Failed: ${result.cause()}")
                    }
                }
        }
    }

    @AfterEach
    fun tearDown(context: VertxTestContext) {
        context.completeNow()
    }

    @Test
    fun testMyApplication(context: VertxTestContext) {
        vertx.createHttpClient().request(HttpMethod.GET, 8080, "localhost", "/")
            .compose { request -> request.send() }
            .onSuccess { response ->
                response.body()
                    .onSuccess { body ->
                        println("Received: ${body.toString("UTF-8")}")
                        context.completeNow()
                    }
                    .onFailure { err ->
                        context.failNow(err)
                    }
            }
            .onFailure { err ->
                context.failNow(err)
            }
    }
}